code/game/objects/items/devices/chameleonproj.dm

```diff
@@ -171,6 +171,84 @@
 	master.disrupt(0)
 	return ..()
 
+/obj/item/chameleon_counterfeiter
+	name = "chameleon counterfeiter"
+	icon = 'icons/obj/device.dmi'
+	icon_state = "cham_counter"
+	flags = CONDUCT
+	slot_flags = SLOT_BELT
+	item_state = "electronic"
+	w_class = WEIGHT_CLASS_SMALL
+	origin_tech = "syndicate=3;magnets=3"
+	var/can_use = TRUE
+	var/saved_item = null
+	var/saved_icon = null
+	var/saved_icon_state = null
+	var/saved_item_state = null
+	var/saved_overlays = null
+	var/saved_underlays = null
+	var/activate_dummy = FALSE
+
+/obj/item/chameleon_counterfeiter/attack_self()
+	matter_toggle()
+
+/obj/item/chameleon_counterfeiter/afterattack(obj/item/target, mob/user, proximity)
+	if(!proximity || !check_sprite(target) || target.alpha < 255 || target.invisibility != 0)
+		return
+	if(!activate_dummy)
+		if(istype(target,/obj/item))
+			playsound(get_turf(src), 'sound/weapons/flash.ogg', 100, 1, -6)
+			to_chat(user, "<span class='notice'>Scanned [target].</span>")
+			saved_item = target.type
+			saved_icon = target.icon
+			saved_icon_state = target.icon_state
+			saved_item_state = target.item_state
+			saved_overlays = target.overlays
+			saved_underlays = target.underlays
+
+/obj/item/chameleon_counterfeiter/proc/check_sprite(atom/target)
+	return (target.icon_state in icon_states(target.icon))
+
+/obj/item/chameleon_counterfeiter/proc/matter_toggle()
+	if(!can_use || !saved_item)
+		return
+	if(activate_dummy)
+		playsound(get_turf(src), 'sound/effects/pop.ogg', 100, 1, -6)
+		matter_deactivate(saved_icon, saved_icon_state, saved_item_state, saved_overlays, saved_underlays)
+		to_chat(usr, "<span class='notice'>You deactivate [src].</span>")
+	else
+		playsound(get_turf(src), 'sound/effects/pop.ogg', 100, 1, -6)
+		var/obj/O = new saved_item(src)
+		if(!O)
+			return
+		to_chat(usr, "<span class='notice'>You activate [src].</span>")
+		matter_activate(O, saved_icon, saved_icon_state, saved_item_state, saved_overlays, saved_underlays)
+
+/obj/item/chameleon_counterfeiter/proc/matter_activate(obj/O, new_icon, new_iconstate, new_item_state, new_overlays, new_underlays)
+	name = O.name
+	desc = "<span class='warning'>It doesn't look quite right...</span>"
+	icon = new_icon
+	icon_state = new_iconstate
+	item_state = new_item_state
+	overlays = new_overlays
+	underlays = new_underlays
+	activate_dummy = TRUE
+
+/obj/item/chameleon_counterfeiter/proc/matter_deactivate(new_icon, new_iconstate, new_item_state, new_overlays, new_underlays)
+	name = initial(name)
+	desc = initial(desc)
+	icon = initial(icon)
+	icon_state = initial(icon_state)
+	item_state = initial(item_state)
+	overlays = initial(overlays)
+	underlays = initial(underlays)
+	activate_dummy = FALSE
+	can_use = FALSE
+	addtimer(VARSET_CALLBACK(src, can_use, TRUE), 3 SECONDS)
+
+/obj/item/chameleon_counterfeiter/attack_self()
```

Comment:
```suggestion
/obj/item/chameleon_counterfeiter/attack_self(mob/living/user)
```
please make this take mobs as a parameter and pass is to the matter_toggle proc to avoid usage of `usr`